build-support/mypy/mypy.ini

```diff
@@ -0,0 +1,40 @@
+[mypy]
+# optionals
+no_implicit_optional = True
+
+# untyped
+check_untyped_defs = False
+disallow_untyped_calls = False
+disallow_untyped_defs = False
+disallow_untyped_decorators = False
+disallow_incomplete_defs = False
+
+# dynamic typing
+disallow_any_unimported = False
+disallow_any_expr = False
+disallow_any_decorated = False
+disallow_any_explicit = False
+disallow_any_generics = False
+disallow_subclassing_any = False
+
+# strictness
+strict_equality = False
+
+# warnings
+warn_unused_ignores = True
+warn_no_return = True
+warn_return_any = True
+warn_redundant_casts = True
+
+# error output
+show_column_numbers = True
+show_traceback = True
+
+# imports
+ignore_missing_imports = True
```

Comment:
This seems like something we should be able to enable today? What violates this?

Reply:
This line is crucial to avoid any of our untyped 3rd party dependencies from causing issues. Without it, MyPy will complain that type information could not be found for libraries like `ansicolors`.

A better approach would be to only turn this value on for every package that is currently missing—MyPy allows you to enumerate each package this should apply to. However, the list was getting really long and the majority of our packages do not have type info, so I went with the simpler global flag.

See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports.

--

I'll link to this in comment.